src/LabLog/Controllers/AdminController.cs

```diff
@@ -22,24 +22,25 @@ public class AdminController : Controller
         private readonly EventModelContext _db;
         private string _user = "user";
         private SchoolService _schoolService;
+        private const string _controllerString = "Admin";

```

Comment:
I'm not entirely sure what you want to achieve with _controllerString, but there are probably better ways than just hard coding it like this.
You could get the name of the current class for example. Depends whether it's important whether changing the controllername should also change _controllerString

Reply:
Its just so that the partial view redirects back to the correct controller. I just want to know whether a view should redirect back to Admin, or Teacher.

Reply:
So, yes, the controllerstring would need to change if the controller's name does. I suppose I could grab the controller from the existing url string.

Reply:
You could do something like `Regex.Match("(?<controllerName>.*)Controller", this.GetType().Name)`
though if you want to go down that path there are a couple of caveats I could help you with

Reply:
Or just do this: https://stackoverflow.com/questions/18248547/get-controller-and-action-name-from-within-controller

Reply:
This seems to work too -> public async Task<IActionResult> Index(string controller)

Reply:
It'd be ugly with multiple things to track though, so RouteData is probably better, if a bit verbose

Reply:
But how is using a constant worse than hard coding the `asp-controller="admin"` in the Views?

Reply:
pretty similar really. I don't like magic string, wherever they come from. At least in the asp-controller visual studio will use intellisense to yell you something has gone wrong

Reply:
Fair enough. Currently its defined once, and passed to SchoolService once in the constructor. Then every ViewModel created by the service includes the appropriate controller. But if I switch to RouteData, then it will have to be defined in every route, and passed to the service separately each time, which seems rather less neat.
Is there a way to detect the controller in its own constructor, and sending it through to the service once?

Reply:
This should do it:
`this.ControllerContext.RouteData.Values["controller"].ToString()`
because the constructor is created for your particular request and route. Try it out anyway. If not we can use 'reflection' to get it from the class name

Reply:
ControllerContext is available in the contructor, but RouteData is null at that point.

Reply:
Yeah fair enough. You can use a method to get it at execution time, or use a regex to grab it from the class name. Which sounds more fun? I'll talk you through it

Reply:
This works in the view file itself:
```<a asp-controller="@this.ViewContext.RouteData.Values["controller"].ToString()"```

Reply:
Interesting, could be dangerous though if you want to render the view from a different controller at some stage